platform/state/platform_state.proto

```diff
@@ -0,0 +1,230 @@
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.state.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.platform.state">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * The current state of platform consensus.<br/>
+ * This message stores the current consensus data for the platform
+ * in network state.
+ *
+ * The platform state SHALL represent the latest round's consensus.<br/>
+ * This data SHALL be used to ensure consistency and provide critical data for
+ * restart and reconnect.
+ */
+message PlatformState {
+    /**
+     * A consensus round.<br/>
+     * The round represented by this state.
+     * <p>
+     * This message SHALL represent the network state after handling of all
+     * transactions reaching consensus in all previous rounds.<br/>
+     * The first state (genesis state) SHALL have a round of 0.<br/>
+     * The first round to handle transactions is defined as round `1`.
+     */
+    uint64 round = 1;
+
+    /**
+     * A consensus timestamp for this state.
+     * <p>
+     * This SHALL be the consensus timestamp of the first transaction in
+     * the current round.
+     */
+    proto.Timestamp consensus_timestamp = 2;
+
+    /**
+     * A version describing the current version of application software.
+     * <p>
+     * This SHALL be the software version that created this state.
+     */
+    SoftwareVersion creation_software_version = 3;
+
+    /**
+     * A number of non-ancient rounds.
+     * <p>
+     * This SHALL be the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 4;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     * <p>
+     * This SHALL be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 5;
+
+    /**
+     * A timestamp for the next scheduled time when a freeze will start.
+     * <p>
+     * If a freeze is not scheduled, this SHALL NOT be set.
+     */
+    proto.Timestamp freeze_time = 6;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 7;
+
+    /**
+     * A consensus node software version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * Otherwise, this SHALL be the _first_ software version that enabled birth
+     * round mode.
+     */
+    SoftwareVersion first_version_in_birth_round_mode = 8;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 last_round_before_birth_round_mode = 9;
```

Comment:
This seems a bit less than ideal; I would think a uint64 is a better choice, but we make it signed just to handle a short-term condition.
I wonder if we couldn't use the fact that `first_version_in_birth_round_mode` is unset as the "flag" for "birth round not enabled" and just have a uint64 here that is `0` if birth round mode is not enabled (and also `0` if birth round is enabled at genesis)?
That would enable us to use a `uint64` here without ambiguity and avoid including a signed value just to handle the temporary condition.

Alternatively we could define "birth round not enabled" as MAX_UNSIGNED, which would be `-1` in the java code anyway (so we have the `uint64` field, but define the flag value in those terms and use the protobuf automatic translation to make the code roughly the same).

Reply:
>Alternatively we could define "birth round not enabled" as MAX_UNSIGNED

Yes. I like this idea. I'll do it this way